### DISCUSSION TITLE
Converting generated types to JSON API spec

### DISCUSSION THREAD

**Discussion by domschab23** [openapi-ts/openapi-typescript#1266] [Q&A]
I'm using https://jsonapi.org/ format to serialize my data on the backend and then deserialize that data on the frontend with https://github.com/olosegres/jsona to have more simplified objects. This package caught my interest to generate types based on my openAPI docs. However, those types are generated for the JSON API format before deserialization. I'm interested in having types match simplified object after deserialization happens. Did anyone experience something similar? Is there a way to account for this with openapi-typescript package? 

Cheers

**Comment by drwpow** [comment]
@duncanbeevers opened a related issue on this here: https://github.com/drwpow/openapi-typescript/issues/1328. Though you’re asking for something slightly-different, I do see a lot of overlap here when it comes to serialization/deserialization.

**Reply by duncanbeevers** [reply]
@domschab23 You may be able to use the `transform()` option in openapi-typescript to generate types matching your deserialized data.

For example, in our schema we have a timestamp field like this:

```
createdAt:
  type: string
  format: date-time
```

By default openapi-typescript will generate `createdAt?: string` (or similar).
However, by using the `transform()` option you can generate `createdAt?: Date`.

I *think* the `transform()` functionality is only available through the nodejs API, but it's quite simple to use.
